src/main/java/net/dv8tion/jda/webhook/WebhookClientBuilder.java

```diff
@@ -62,6 +65,29 @@ public WebhookClientBuilder(final long id, final String token)
     }
 
     /**
+     * Creates a new WebhookClientBuilder with the provided webhook URL
+     *
+     * @param  url
+     *         The URL of the webhook. May be directly copied from Discord's UI
+     *         <br>Example: {@code https://discordapp.com/api/webhooks/123456789012345678/my-webhook-token}
+     *         <br>This constructor also parses URLs pointing to subdomains of {@code discordapp.com}
+     *
+     * @throws java.lang.IllegalArgumentException
+     *         If the provided URL is {@code null}
+     *         or is incorrectly formatted
+     */
+    public WebhookClientBuilder(@Nonnull String url)
+    {
+        Matcher matcher = WEBHOOK_PATTERN.matcher(url);
+        if(!matcher.find()) {
```

Comment:
Why shouldn't this be a proper regex to ensure the URL is actually a discord url, then `pattern.matches(url)`

Reply:
Why the `pattern.matches(url)`? If #find() fails it wouldn't match anyways

Reply:
We are looking for a proper URL not some 2k char message with some url in it.

Reply:
Also, do you want me to verify if the String indeed is a HTTP(S) URL, or should we ignore the schema? Eg do we consider `discordapp.com/api/webhooks/370264727500095498/token` valid?

Reply:
I would suggest to make it optional and use the matches method. `(https?://)?(www\.)?.......`